SPECS-CROSS/cross-gcc/cross-gcc.spec

```diff
@@ -0,0 +1,161 @@
+%global debug_package %{nil}
+%global __os_install_post %{nil}
+
+# Globals which should be in a macro file.
+# These should be set programatically in the future.
+%global _host_arch      x86_64
+%global _target_arch    aarch64
+
+%global _tuple          %{_target_arch}-%{_vendor}-linux-gnu
+%global _cross_name     %{_target_arch}-%{_vendor}-linux-gnu
+
+# Folders which should be in our macro file
+%global _opt                /opt/
+%global _crossdir           /opt/cross/
+
+# Generally we include '/usr' in most paths.
+# Can we also use '/usr' for our paths? This will bring us in line with the
+# %%configure macro which sets these.
+%global _bindir            /bin
+%global _sbindir           /sbin
+%global _libdir            /lib
+%global _lib64dir          /lib64
+%global _libexecdir        /libexec
+%global _datadir           /share
+%global _docdir            /share/doc
+%global _includedir        /include
+%global _infodir           /share/info
+%global _mandir            /share/man
+%global _oldincludedir     /include
+
+# If we want our cross compile aware packges to also support native, we
+# need logic to switch modes something like this:
+%if "%{_target_arch}" != "%{_host_arch}"
+%global _cross_prefix       %{_crossdir}%{_tuple}/
+%global _cross_sysroot      %{_crossdir}%{_tuple}/sysroot/
+%global _cross_includedir   /usr/%{_host}/%{_tuple}/include/
+%global _cross_infodir      %{_crossdir}%{_tuple}/share/info
+%global _cross_bindir       %{_tuple}/bin
+%global _cross_libdir       %{_tuple}/lib
```

Comment:
Some of these macros (`_cross_libdir` for instance) seem unused in this spec. Also, we do we re-define them in each spec? I'm curious, if we could keep them all in one place in a separate "cross-macros" package and only manipulate the value of `_tuple` depending on the architecture we build for.